app/models/concerns/blacklight/user.rb

```diff
@@ -33,8 +34,21 @@ def existing_bookmark_for(document)
   ##
   # @return [String] a user-displayable login/identifier for the user account
   def to_s
-    return email if respond_to?(:email)
+    string_display_key = self.class.string_display_key
+    return send(string_display_key) if respond_to?(string_display_key)
 
     super
   end
+
+  module ClassMethods
+    def string_display_key(attr_name = nil)
```

Comment:
Can you use `class_attribute :string_display_key, default: :email` ?   That means we don't need all this code.

Aside from that I think this approach works.

Reply:
I think part of @cbeer's reservation initially is that the `:email` attribute only exists if you have set up Devise, which is not strictly required. But I think you're right about using the class attribute - I could just have the generated code from Devise setup be `string_display_key ||= :email` (we don't want to assign unconditionally because the generated code shouldn't stomp on mixin assignments). I'll make that change!